pkg/prowgen/prowgen.go

```diff
@@ -394,10 +394,22 @@ func injectCapabilities(labels map[string]string, capabilities []string) {
 	}
 }
 
-func injectArchitectureLabels(g *prowJobBaseBuilder, imagesConfig []cioperatorapi.ProjectDirectoryImageBuildStepConfiguration) {
+// injectCapabilitiesForImgJobs injects capabilities for image build jobs
+// If the image build job has capabilities, it will use them and skip architecture labels
+// If the image build job does not have capabilities, it will use architecture labels
+func injectCapabilitiesForImgJobs(g *prowJobBaseBuilder, imagesConfig []cioperatorapi.ProjectDirectoryImageBuildStepConfiguration) {
+	capabilities := false
 	for _, imageConfig := range imagesConfig {
-		for _, arch := range imageConfig.AdditionalArchitectures {
-			g.WithLabel(fmt.Sprintf("capability/%s", arch), arch)
+		for _, capability := range imageConfig.Capabilities {
+			g.WithLabel(fmt.Sprintf("capability/%s", capability), capability)
+			capabilities = true
+		}
+	}
+	if !capabilities {
+		for _, imageConfig := range imagesConfig {
+			for _, arch := range imageConfig.AdditionalArchitectures {
```

Comment:
If the fields are not mutually exclusive, we could just merge them. Otherwise, we need to validate it.

Reply:
right now we analyze capabilities and if empty we analyze additional arches. I will remove additional arches field within 2-3 days